src/system.rs

```diff
@@ -344,7 +345,7 @@ pub(crate) fn hot_reload_style_sheets(
         if let AssetEvent::Modified { id } = evt {
             q_sheets
                 .iter_mut()
-                .filter(|sheet| sheet.handle().id() == *id)
+                .filter(|sheet| sheet.handles().iter().any(|h| h.id() == *id))
```

Comment:
If the `Node` has 3 sheets and any of them is changed, all three sheets will be refreshed, which doesn't seems the intended thing to do.

This is out of scope of this PR, but ideally we should be able to choose which sheet to refresh, something like `sheet.refresh_id(id)`

If you agree, we could open an issue and add a todo, to keep track of this later on

Reply:
I may be misunderstanding you, but in this case, `refresh()` is being called on the singular style sheet list component, not individual asset handles. The component itself is only being refreshed once, not the individual asset handles.

The component is being refreshed once if any of the asset handles within it are changed. The only instance where the component is refreshed multiple times is when multiple assets are reload at the same time.

Reply:
Yes, you understood right, and this is a problem. Let's say we have three css files loaded in a single `StyleSheet` component:
1. theme.css
2. font.css
3. layout.css

If I change something inside `theme.css`, the asset change watcher will trigger an event saying this asset was changed and will send an event to us. In this case, it will match the `iter().any()` and `sheet.refresh()` will be called. Fine our component will be refreshed  and everything will be applied.

But the problem is all three style sheets will be reapplied, not only `theme.css`, since `sheet.refresh()` triggers the component change detection as a whole. So if somehow `layout.css` or `font.css` is a big stylesheet or something like that, it will be applied again, even if there is no modification on the asset.

This isn't a big problem, since it will run only once, not every frame, but it is something to keep in mind and if we can, we should improve later on. The solution would be to know which asset id (or ids) triggered the `sheet.refresh()` and only reapply those. But as I said, this should be out of scope of this PR.

Reply:
Oh, yes, I see what you're referring to now, and I agree completely.

Yes, that is out of the scope of this PR, like you said, and will require a much more in-depth implementation of merging. The current implementation relies on later layers overriding earlier layers to account for rule priority. If twop layers implement different rules for the same selection, the later one will overwrite the early one and is a cheap way to work around priority values.

If we were to refresh and reapply an earlier sheet, it would suddenly take priority and overwrite a higher layer. The only way to prevent this would be to do a far more detailed analysis of each rule and skip applying rules to components that shouldn't apply and all that. In the best case, this would still be just as expensive an operation as the current implementation, but drastically increase the complexity of the code.

Plus, it's debatable whether this analysis is really needed, since hot-reloading can only occur in dev builds, anyway.

Ideally, a rework of how rules are implemented would be needed for a more proper merging of stylesheets rather than just layering them. It would make implementation of traits such as `important!` possible. Though this would require individual rules to be tracked back to their original style sheet assets so they can be ignored when re-merging all the style sheets.

No matter how it's coded, reloading one asset will still require work to be done on all linked assets, if for no other reason than handling rule collisions.